apps/judicial-system/backend/src/app/modules/case/case.controller.ts

```diff
@@ -231,9 +232,33 @@ export class CaseController {
       update.courtRecordSignatureDate = null
     }
 
-    // The only roles that can appeal a case are prosecutor roles
+    // TODO: Consider changing the names of the postponed appeal date variables
+    // as they are now also used when the case is appealed in court
     if (states.appealState === CaseAppealState.APPEALED) {
+      // The only roles that can appeal a case here are prosecutor roles
       update.prosecutorPostponedAppealDate = nowFactory()
+    } else if (
+      // Handle appealed in court
+      !theCase.appealState &&
+      [CaseTransition.ACCEPT, CaseTransition.REJECT].includes(
+        transition.transition,
+      ) &&
+      (theCase.prosecutorAppealDecision === CaseAppealDecision.APPEAL ||
+        theCase.accusedAppealDecision === CaseAppealDecision.APPEAL)
+    ) {
+      if (theCase.prosecutorAppealDecision === CaseAppealDecision.APPEAL) {
+        update.prosecutorPostponedAppealDate = nowFactory()
+      } else {
+        update.accusedPostponedAppealDate = nowFactory()
+      }
+
+      const autoStates = transitionCase(
+        CaseTransition.APPEAL,
+        states.state ?? theCase.state,
+        states.appealState ?? theCase.appealState,
+      )
+      update.state = autoStates.state
```

Comment:
Do we need to do this here? transitionCase shouldn't change the case state when called with APPEAL, and we've already set the new case state in line 223.

Reply:
I did not want to depend on knowledge about the transition method, it may change in the future.